lib/structuraid_core/engineering/vector.rb

```diff
@@ -1,34 +1,66 @@
 module StructuraidCore
   module Engineering
     class Vector < Base
-      attr_accessor :value_x, :value_y, :value_z
+      attr_accessor :value_i, :value_j, :value_k
 
-      def self.with_value(value:, direction:)
-        new(
-          value_x: value.to_f * direction[0],
-          value_y: value.to_f * direction[1],
-          value_z: value.to_f * direction[2]
-        )
+      class << self
+        def from_location(location:)
+          from_matrix(location.to_matrix)
+        end
```

Comment:
also just a nitpick. I think it's pretty obvious what kind of argument these `from_*`, so I think there is no need to use keyword arguments for these ones. Let me know what you think
```suggestion
        def from_location(location)
          from_matrix(location.to_matrix)
        end
```

Reply:
Also I'm seeing below that you are actually defining `from_matrix` with a keyword argument, so this method will fail because you are passing it a normal argument. Is there a spec for this method?

Reply:
yeah, will check it